test/machine/main.go

```diff
@@ -136,3 +158,14 @@ func captureOutput(f func()) string {
 	writer.Close()
 	return <-out
 }
+
+// MemorySink implements zap.Sink by writing all messages to a buffer.
```

Comment:
hmm iirc there's a zap debug pkg that might be better?

Reply:
sorry, bad rebase. this file should be removed now